frontend/src/components/navbar/Navbar.tsx

```diff
@@ -1,31 +1,67 @@
-import React, { useContext } from "react";
+import React, { useContext, useState } from "react";
 import { Link } from "react-router-dom";
 import { LoginContext } from "../login/LoginManager";
 import "./Navbar.scss";
+import Hamburger from "hamburger-react";
 
 export const Navbar: React.FunctionComponent = () => {
   const loginContext = useContext(LoginContext);
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  function toggleMenu() {
+    if (isExpanded) {
+      setIsExpanded(false);
+    }
+  }
```

Comment:
You're missing an `else`! This will only ever close the menu, but can't open it!

You can write this very efficiently as:
```suggestion
  function toggleMenu() {
    setIsExpanded(!isExpanded);
  }
```

Reply:
This is a little advanced note, so skip this if you don't understand it...

React's `setState()` methods take the new value of the state as a parameter, which is how we've been using them: `setCount(2)` or whatever. But, instead of passing it the new value, you can instead pass it "a function that takes in the old value and returns the new value".

I don't actually know if this is more efficient or something, but this means instead of writing, for example:
```
setCount(count + 1);
```
you can instead write
```
setCount(oldCount => oldCount + 1);
```

Researching this led me down a React rabbit hole that even I don't understand, though, so probably just one to shelve riiiiight in the back of your minds for now!